docker-image/Dockerfile

```diff
@@ -40,23 +42,31 @@ RUN apt-get update \
         netcat \
         \
         docker-ce-cli \
-        python3.7 \
-        python3-pip \
+        python3.{7,10}{,-distutils} \
         rubygems \
 && apt-get clean \
 \
-&& python3.7 -m pip install --upgrade \
-        lxml pip setuptools wheel \
-&& python3.7 -m pip install --upgrade \
-        ansible-lint \
-        ansible-lint-junit>=0.16 \
-        boto3 \
-        jinja2 \
-        pylint \
-        pylint-fail-under \
-        pytest \
-        python-json-logger \
-        ruamel.yaml \
+&& for version in 3.7 3.10 ;\
```

Comment:
I would avoid looping since it forces the same dependencies for all Python versions (and all supported Epiphany branches).

I would pre-install Python packages only for the latest Python (for develop branch) since builds for other branches are run much less often so we accept they can take more time. They will install what they need via pipeline's tasks on the first run on agent (or after agent's container was restarted).

Reply:
That would force us to manually edit most old pipelines, which would consume more time than building this image. I understand that installing all dependencies for both versions of the interpreter may seem sub-optimal but it is ultimately a time saver. Consider that this build is not run often either.

Reply:
The good practice is that a pipeline installs all dependencies it needs - ideally it should be agent "agnostic" (portable) as much as possible. But for now we can keep the loop as discussed - adding a comment that this is not the final solution :)